pkg/storage/bucket/s3/bucket_client.go

```diff
@@ -153,8 +154,18 @@ func (b *BucketWithRetries) Exists(ctx context.Context, name string) (exists boo
 }
 
 func (b *BucketWithRetries) Upload(ctx context.Context, name string, r io.Reader) error {
+	// Convert Reader to Seeker
+	var buf bytes.Buffer
+	if _, err := buf.ReadFrom(r); err != nil {
```

Comment:
This will read the whole file in memory...

I think most of the cases we are reading from a file and the file is never entirely in memory when uploading the blocks, right?

Reply:
Should we just seek if possible? Something like:

```
			seeker, ok := reader.(io.Seeker)
			if ok {
				offset, err := seeker.Seek(0, io.SeekCurrent)
			}
```

Reply:
Updated the logic to only retry if `r` is also `Seeker`